### PR TITLE
fix!: EXPOSED-360 Storing ULong.MAX_VALUE in ulong column not working

### PR DESCRIPTION
**Pull request by joc-a**
### **The issue**:
When attempting to store `ULong.MAX_VALUE` in a `ulong` column, retrieving it results in -1 and not `ULong.MAX_VALUE` for all database dialects (except for MySQL and MariaDB). The reason is that we invoke `.toLong()` on the `ULong` value before we store it in the database (in `notNullValueToDB` in `ULongColumnType`) and that gets the -1 from [here](https://github.com/JetBrains/kotlin/blob/0938b46726b9c6938df309098316ce741815bb55/libraries/stdlib/unsigned/src/kotlin/ULong.kt#L27) because of the [implementation](https://github.com/JetBrains/kotlin/blob/0938b46726b9c6938df309098316ce741815bb55/libraries/stdlib/unsigned/src/kotlin/ULong.kt#L323) of `ULong.toLong()`.

### **The fix**:
Before this fix, the maximum value that could be stored with `ULongColumnType` for all database dialects (except for MySQL and MariaDB) was `Long.MAX_VALUE`. This is because the `ULong` value was being converted to `Long` in the `notNullValueToDB` function, restricting the range to that of `Long`. Instead, it is now converted to a `BigInteger` for PostgreSQL, a `Long` for PostgreSQLNG (with a range check), and a `String` for the rest of the database dialects. The `ulong` column can now store `ULong.MAX_VALUE` for H2 (excluding H2_PSQL), Oracle, SQLite and SQL Server.

### **Breaking change**:
`ulongType()` is now NUMERIC(20) instead of BIGINT for H2 (excluding H2_PSQL), SQLite, and SQL Server to allow storing the full range of `ULong`, including `ULong.MAX_VALUE`.

### **Limitations**:
- For PostgreSQL, it is not possible to change `ulongType()` to NUMERIC(20) because that would cause a mismatch when it references an auto-incrementing ulong column of type BIGSERIAL. It's also not possible to change the `autoIncType` to NUMERIC(20) either, because PostgreSQL does [not](https://www.postgresql.org/docs/current/datatype-numeric.html#:~:text=The%20data%20types%20smallserial%2C%20serial%20and%20bigserial%20are%20not%20true%20types%2C%20but%20merely%20a%20notational%20convenience%20for%20creating%20unique%20identifier%20columns%20(similar%20to%20the%20AUTO_INCREMENT%20property%20supported%20by%20some%20other%20databases).) support the AUTO_INCREMENT property. So the maximum value that can be stored is `Long.MAX_VALUE`.
- For SQLite, it is [not possible](https://www.sqlite.org/autoinc.html#:~:text=Because%20AUTOINCREMENT%20keyword%20changes%20the%20behavior%20of%20the%20ROWID%20selection%20algorithm%2C%20AUTOINCREMENT%20is%20not%20allowed%20on%20WITHOUT%20ROWID%20tables%20or%20on%20any%20table%20column%20other%20than%20INTEGER%20PRIMARY%20KEY.) to have auto-increment on a column of type other than INTEGER PRIMARY KEY, so the maximum value that can be stored using a `ulong` auto-increment column is the same as that of a `long` column.

### CODE REVIEW

**Comment by joc-a** [comment]
@e5l @bog-walk I want to run more tests on some corner cases with references before I merge this, so you can ignore this PR for now.

**Comment by joc-a** [comment]
Ready for review @bog-walk @e5l

File: exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/vendors/DataTypeProvider.kt [hunk 1]
```diff
@@ -55,13 +55,13 @@ abstract class DataTypeProvider {
     open fun longType(): String = "BIGINT"
 
     /** Numeric type for storing 8-byte unsigned integers. */
-    open fun ulongType(): String = "BIGINT"
+    open fun ulongType(): String = "NUMERIC(20)"
```
Comment (e5l) [inline exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/vendors/DataTypeProvider.kt:58]: could you tell me if it's a breaking change?

**Review by e5l — APPROVED** [review]

File: exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/vendors/DataTypeProvider.kt (hunk 1 above)
Comment (joc-a) [reply exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/vendors/DataTypeProvider.kt:58]: Yes, it is.

**Review by bog-walk — APPROVED** [review]
Could a line be added to `Table.ulong()` KDocs, specifying that the column will accept up to and including the max value for all dialects except PG?
